src/business_logic/transaction/objects/v2/declare_v2.rs

```diff
@@ -211,8 +224,21 @@ impl InternalDeclareV2 {
         )
     }
 
+    fn store_casm_class<S: Default + State + StateReader + Clone>(
+        &mut self,
+        casm_class: CasmContractClass,
+        state: &mut S,
+    ) -> Result<(), StateError> {
+        let compiled_class_hash = self.compiled_class_hash.to_le_bytes();
+        let class_hash = self.hash_value.to_le_bytes();
```

Comment:
Is there any reason why the conversion to bytes happens outside of the `state`'s `set_compiled_class_hash()`? This way it looks like it is possible for someone to insert little-endian bytes and maybe have some other part of the program that gets or sets big-endian bytes introducing some issues/ambiguities. 
If there is no specific reason, the `state` interface should probably enforce the structure of the input (not for the scope of this PR, but maybe as part of #472)